### ISSUE TITLE
[no-void-expression] Invalid entry on ROADMAP

### ISSUE THREAD

**Issue by manuth** [typescript-eslint/typescript-eslint#1871]
Hey guys
I just noticed that the alternative listed for the `no-void-expression`, which, according to the `ROADMAP.md`-file, is `no-void` doesn't do the same thing like it's `tslint`-counterpart.

While `tslint`'s `no-void-expression` tries to prevent users from assigning, returning or using the result of a void expression, `eslint`'s `no-void` bans the `void`-keyword.

## `tslint`'s `no-void-expression`
***tslint.json:***
```json
    "rules": {
        "no-void-expression": [
            true,
            "ignore-arrow-function-shorthand"
        ]
    }
```
### _Example of bad code for `tslint`'s `no-void-expression`_
***tes.ts:***
```ts
function test(): void { }
console.log(test()); // bad
let x = test(); // bad
function test2() { return test(); } // bad
```

### _Example for good code for `tslint`'s `no-void-expression`_
***test.ts:***
```ts
function test(): void { }
test(); // good
```

### _Example of bad code for `eslint`'s `no-void`_
```ts
function test(): void
{
    return void 0; // bad
}
```
### _Example of good code for `eslint`'s `no-void`_
```ts
function test(): void
{
    return undefined; // good
}

function test2(): void
{
    return; // good
}
```

**Comment by bradzacher** [comment]
NGL the tslint rule is terribly named. No doubt that's why the original author of the roadmap got confused. The fact nobody's picked up on it in 16 months (https://github.com/bradzacher/eslint-plugin-typescript/pull/228) shows that either nobody really uses the rule, or nobody understands the difference...

Happy to accept a PR to fix this!

**Comment by manuth** [comment]
I got confused myself, tbh - I had to fiddle around for about 20 minutes to find out what it's actually made for 😅 
That's why I tried to provide some sort of example code
